libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -851,11 +865,27 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 				break;
 			}
 			case DataLocation::Memory:
-				solUnimplementedAssert(false, "");
-				break;
 			case DataLocation::CallData:
-				solUnimplementedAssert(false, "");
+			{
+				string const memAddress =
+					(arrayType.location() == DataLocation::Memory ?
+						m_utils.memoryArrayIndexAccessFunction(arrayType) :
```

Comment:
arrayIndexAccessFunction?

Reply:
You prefer one function handling all cases?

Reply:
I would say yes: As long as calldata arrays are not yet implemented we do not know how different the implementation will be.

Reply:
Just to be clear: You don't want to merge storage + memory array index in one function though?

Reply:
OK right, then maybe just leave it as it is.